lib/BasePlugin/plugin.rb

```diff
@@ -30,8 +31,15 @@ def register
   # Returns Ruby Hash of report data
   def report(resource = {node: '*'}, start_time = Time.now - (30 * SECONDS_IN_DAY),
              end_time = Time.now)
+    log.error StandardError.new(
+      "start_time and end_time should be Time objects"
```

Comment:
We might want to name-space these error messages, something like "#{class_name}: start_time and end_time should be Time objects".
I'm not sure the programmatic  to get a class name but that way we can more accurately debug the code.